### PR TITLE
Upgrade hapi

### PR DESCRIPTION
**Pull request by tomreeves**
Change includes:

- Hapi upgrade to v18.4.0 (v19.x.x requires node 12.x.x, which will be coming as an iteration)
- Require Node v10.18.1 which will unblock negating certain security vulnerabilities. Buys us another ~12 months or so.
- Remove Boom / @hapi/boom (not used)

### CODE REVIEW

File: src/server/index.js [hunk 1]
```diff
@@ -55,7 +55,7 @@ class Server {
 
 export const registerPlugins = async ({ server, config }) => {
   // Handles static files and proxying
-  const plugins = [require('inert'), require('h2o2')]
+  const plugins = [require('@hapi/inert'), require('@hapi/h2o2')]
```
Comment (industriousparadigm) [inline src/server/index.js]: Would it make sense to use the import syntax fully in this file, i.e. getting rid of require entirely?

File: src/server/handlers/redirect.js [hunk 2]
```diff
@@ -6,7 +6,9 @@ import { log } from '../utilities/logger'
 
 const setRedirects = (server, redirects) => {
   server.ext('onPreHandler', async (request, h) => {
-    const redirect = redirects[request.url.pathname.toLowerCase()]
+    const redirect =
+      redirects[decodeURIComponent(request.url.pathname.toLowerCase())]
```
Comment (industriousparadigm) [inline src/server/handlers/redirect.js]: Interesting, why this change?

File: src/server/index.js (hunk 1 above)
Comment (tomreeves) [reply src/server/index.js]: Why do you think that would be better?

File: src/server/handlers/redirect.js (hunk 2 above)
Comment (tomreeves) [reply src/server/handlers/redirect.js]: Upgrading Hapi is causing for some random encodings - need to investigate why as it's bugging me. IMO it's safer to decodeURIComponent anyway as it's being used as a key to find the redirect path.

**Comment by romulodl** [comment]
I think you will have to close and open a new PR with these changes to be picked up by Circle CI

**Comment by romulodl** [comment]
:(

File: src/server/index.js (hunk 1 above)
Comment (industriousparadigm) [reply src/server/index.js]: Consistency I guess. Apart from that I don't see any other benefit (or drawback)

Comment (tomreeves) [reply src/server/index.js]: We could, but the hapi-require-https module would still need to use require as it's optional. Good spot but will keep it as is for now.

**Comment by tomreeves** [comment]
> :(

Damn you

File: src/config/babel.js [hunk 3]
```diff
@@ -33,17 +33,24 @@ module.exports = (target = 'node') => {
...
+
       // loadable(() => import('./component.js'))
       require.resolve('@loadable/babel-plugin'),
+
       // Add support for async/await
-      require.resolve('@babel/plugin-transform-runtime')
+      require.resolve('@babel/plugin-transform-runtime'),
+
+      //const safe = new obj?.qux?.baz(); // undefined
+      require.resolve('@babel/plugin-proposal-optional-chaining')
```
Comment (industriousparadigm) [inline src/config/babel.js]: 😍🎉 This changes everything!!! 👋🏻 `idx`
